junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertLinesMatch.java

```diff
@@ -37,6 +38,15 @@ private AssertLinesMatch() {
 	private final static int MAX_SNIPPET_LENGTH = 21;
 
 	static void assertLinesMatch(List<String> expectedLines, List<String> actualLines) {
+		assertLinesMatch(expectedLines, actualLines, () -> "");
+	}
+
+	static void assertLinesMatch(List<String> expectedLines, List<String> actualLines, String message) {
+		assertLinesMatch(expectedLines, actualLines, () -> message);
+	}
+
+	static void assertLinesMatch(List<String> expectedLines, List<String> actualLines,
+			Supplier<String> messageSupplier) {
```

Comment:
Please rework this to follow the `Object messageOrSupplier` implementation _pattern_ used in `AssertThrows`.

Doing so prevents unnecessary creation of lambdas.

Reply:
Sure.

Reply:
Thanks!

And by the way, in case that "pattern" does make sense there, I think you get the gist of what I'm asking.